Library/Navigation.swift

```diff
@@ -596,6 +591,12 @@ private func parsedParams(url: URL, fromTemplate template: String) -> RouteParam
     .components(separatedBy: "/")
     .filter { $0 != "" && !$0.hasPrefix("?") }
 
+  // if we're parsing against the '/' emailClick template and this is a recognized email host
+  // return the expected params for that route to be resolved
+  if templateComponents.isEmpty && isRecognizedEmailHost {
```

Comment:
This is the key logic, once we know we're on the empty `/` route template, and this is an email link we can just return an empty `RouteParams.object([:])` so that it's seen as a match and our parser will resolve it against the `emailClick(_ params: RouteParams)` function so that it returns `Navigation.emailClick`.